src/IB/IIMethod.cpp

```diff
@@ -1573,17 +1626,18 @@ IIMethod::computeFluidTraction(const double data_time, unsigned int part)
 
     if (MathUtilities<double>::equalEps(data_time, d_current_time))
     {
-        X_vec = d_X_current_vecs[part];
+        X_vec = getMeshCoordinates(d_use_current_mesh_configuration_for_interactions, "current", part);
     }
     else if (MathUtilities<double>::equalEps(data_time, d_half_time))
     {
-        X_vec = d_X_half_vecs[part];
+        X_vec = getMeshCoordinates(d_use_current_mesh_configuration_for_interactions, "half", part);
     }
     else if (MathUtilities<double>::equalEps(data_time, d_new_time))
     {
-        X_vec = d_X_new_vecs[part];
+        X_vec = getMeshCoordinates(d_use_current_mesh_configuration_for_interactions, "half", part);
     }
```

Comment:
If `getMeshCoordinates` took `data_time` as an argument, you could collapse all of this to a single line:
```
    X_vec = getMeshCoordinates(data_time, part);
```